### DISCUSSION TITLE
Can the model be fine-tuned using locally annotated data?

### DISCUSSION THREAD

**Discussion by Merle-h** [google/cameratrapai#47] [Q&A]
Hello! While using the model for species identification and classification, I noticed that some species were not accurately recognized — for example, certain bird species were only labeled as “bird,” and there were also some misclassifications (such as the Siberia weasel being identified as the American mink, and the Amur hedgehog being labeled as a nutria).

I exported the model’s prediction results and corrected some of the labels using an image annotation platform (Label Studio). I would like to ask whether it is possible to fine-tune or retrain the model locally using these corrected annotations. If so, could you please provide some guidance or step-by-step instructions on how to do this? Thank you!

**Comment by agentmorris** [comment]
There are several tools available for fine-tuning SpeciesNet, and I'll list the ones I'm aware of below, but I would advise fine-tuning only as a last resort.  In general, it's not possible to fine-tune a computer vision model on just a subset of the species you're interested in, while keeping others from the original class set.  I.e., the process of fine-tuning is mostly the same as training a complete model from scratch; the advantage of fine-tuning is in reducing the volume of training data for a new class set, not in reducing the complexity of the process.

So before fine-tuning, I would strongly consider just doing some minor remapping of the outputs to the labels/taxa you care about.  In your example, it's straightforward to map "American mink" to "Siberian weasel", or in fact if you only have one weasel species in your study area, to map all weasels to "Siberian weasel".  We don't have any tools in this repo to do this kind of mapping right now, but I do have a function in the MegaDetector repo that does this, called "restrict_to_taxa_list" ([documentation](https://megadetector.readthedocs.io/en/latest/postprocessing.html#megadetector.postprocessing.classification_postprocessing.restrict_to_taxa_list), [code](https://github.com/agentmorris/MegaDetector/blob/18c97e9c7dd43679fb31e1a3fda8ff98d4e4e6cb/megadetector/postprocessing/classification_postprocessing.py#L1102)).  Basically you create a .csv file that includes taxa in the SpeciesNet taxonomy and the names you want to use for them, and that function handles mapping child taxa to parent taxa.  I made a little web app [here](https://dmorris.net/speciesnet-taxonomy-mapper/) to help create those .csv files.  I won't say it's super-duper-easy to use this approach, but it's <i>gobs</i> easier than creating a training dataset and fine-tuning.

All that said, if you want to fine-tune, here are a couple of repos/notebooks that facilitate fine-tuning:

* [SpeciesNet Convert](https://github.com/pjbull/speciesnet-convert)
* [Finetuning SpeciesNet](https://github.com/PetervanLunteren/AddaxAI/blob/main/markdown/finetune-speciesnet-simple.md)

Hope that helps!  Closing this discussion for now, but if this doesn't adequately answer your question, LMK and I'll re-open.
